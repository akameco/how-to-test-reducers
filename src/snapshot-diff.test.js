// @flow
import snapshotDiff from 'snapshot-diff'
import reducer, { initialState } from '.'

test('init', () => {
  expect(
    // $FlowFixMe
    snapshotDiff(initialState, reducer(initialState, { type: '@@INIT' }))
  ).toMatchSnapshot()
})

test('inc', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, { type: 'inc' }))
  ).toMatchSnapshot()
})

test('dec', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, { type: 'dec' }))
  ).toMatchSnapshot()
})

test('setText', () => {
  expect(
    snapshotDiff(
      initialState,
      reducer(initialState, { type: 'setText', payload: 'test' })
    )
  ).toMatchSnapshot()
})
