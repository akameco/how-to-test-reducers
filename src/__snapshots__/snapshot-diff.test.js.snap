// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dec 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"count\\": 0,
+   \\"count\\": -1,
    \\"text\\": \\"\\",
  }"
`;

exports[`inc 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"count\\": 0,
+   \\"count\\": 1,
    \\"text\\": \\"\\",
  }"
`;

exports[`init 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`setText 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
    \\"count\\": 0,
-   \\"text\\": \\"\\",
+   \\"text\\": \\"test\\",
  }"
`;
